### PR TITLE
Comment syntax highlighting (Coderay to Rouge transition)

### PR DESCRIPTION
**Pull request by pk12**
# Description
Hello again everyone,

During my preparation, I also implemented this issue so I thought to myself why not merge this too if it's good enough.

This issue introduces a new Markdown Syntax Highlighting Gem, Rouge. Rouge was introduced in order to replace Coderay that according to #6673, caused some problems.

This is a Work In Progress Pull Request.

# Questions

- Is this issue still Relevant? __Answered__
- Should I completely remove the Coderay implementation from the `MarkdownRenderer#block_code` method and remove the gem too?
- The theme (GitHub) that I have currently added does not highlight code that doesn't specify the language like 
`var num = 3` but, from what I read it should be possible to support highlighting for these cases too. Should I try to add it?

# Screenshots

## Coderay Ruby Code
Coderay

## Rouge Ruby code with Github Theme
Rouge

### CODE REVIEW

**Comment by dmarcoux** [comment]
Hey @pk12, we don't use tags like `[draft]` or `[webui]` for PRs and commits. It's really not a big deal, I just wanted to let you know. It's something we used to do about a year ago, so I guess this is where you saw this in the commit history.

**Comment by pk12** [comment]
Hey @dmarcoux, thank you for letting me know and for updating the titles!

File: src/api/app/assets/stylesheets/webui/rouge.css.erb [hunk 1]
```diff
@@ -0,0 +1 @@
+<%= Rouge::Themes::Github.render(scope: '.highlight') %>
```
Comment (dmarcoux) [inline src/api/app/assets/stylesheets/webui/rouge.css.erb:1]: There should be a new line at the end of this file. This also applies to other files in this PR.

This is a short explanation on why it's needed: https://thoughtbot.com/blog/no-newline-at-end-of-file.

Comment (pk12) [reply src/api/app/assets/stylesheets/webui/rouge.css.erb:1]: Wow, that's good to know. Checking the files now.

**Comment by pk12** [comment]
@dmarcoux I have left some questions on the PR description in order to complete the issue, can please you (or someone from the team) look into the questions when you get the time?

**Comment by vpereira** [comment]
Hi @pk12 thank you for your contribution.

If you don't mind please take a look on this guide https://chris.beams.io/posts/git-commit/

Basically we don't write commit messages on the past tense but using imperative form.  So instead of "Implemented X, Y" we write  "Implements X, Y"

Regarding Rouge x Coderay, if i remember right we evaluated back in the days both solutions and for some reason (that I don't remember why) we decided to go with Coderay.  I'm not sure if the specs that we have are enough so some detailed test plan would be necessary here.

**Comment by pk12** [comment]
> Hi @pk12 thank you for your contribution.
> 
> If you don't mind please take a look on this guide https://chris.beams.io/posts/git-commit/
> 
> Basically we don't write commit messages on the past tense but using imperative form. So instead of "Implemented X, Y" we write "Implements X, Y"
> 
> Regarding Rouge x Coderay, if i remember right we evaluated back in the days both solutions and for some reason (that I don't remember why) we decided to go with Coderay. I'm not sure if the specs that we have are enough so some detailed test plan would be necessary here.

Hi @vpereira,

Thank you for your response. Regarding the commit message tense, I am already aware of it but I had implemented the commits before learning about it and didn't have the time to update the messages.
That's good to know, I will close the PR.

**Comment by vpereira** [comment]
Hi @pk12 you don't need to close the PR. We just have to make sure that we test it well. Thank you for your contribution :+1:

**Comment by pk12** [comment]
@vpereira I will try to think of more test cases tonight and I will get back at you tomorrow. If you have any suggestions please let me know :smile:

**Comment by vpereira** [comment]
Hey @pk12 !

You don't have to worry about the (manual) test cases. Who are reviewing the PR can take a look on that. Basically we should make sure that the whole areas where we render markdown or code (using after this PR, `rogue`) is being tested and no regression was introduced. Probably we will need some manual testing, but that we do while validating the PR! For now probably, just rewrite the git messages would be awesome

**Comment by pk12** [comment]
> Hey @pk12 !
> 
> You don't have to worry about the (manual) test cases. Who are reviewing the PR can take a look on that. Basically we should make sure that the whole areas where we render markdown or code (using after this PR, `rogue`) is being tested and no regression was introduced. Probably we will need some manual testing, but that we do while validating the PR! For now probably, just rewrite the git messages would be awesome

@vpereira I see :cool: . I will make sure to update the commit messages later today!

**Comment by Conan-Kudo** [comment]
Rouge doesn't have an RPM spec file syntax highlighting, is someone going to fix that? https://github.com/rouge-ruby/rouge/issues/754

**Comment by vpereira** [comment]
> Rouge doesn't have an RPM spec file syntax highlighting, is someone going to fix that? [rouge-ruby/rouge#754](https://github.com/rouge-ruby/rouge/issues/754)

@Conan-Kudo thank you for your heads-up.

**Comment by vpereira** [comment]
Another point:

Running a benchmark framework for syntax highlighting libs https://github.com/rubychan/shootout, looks like coderay is much faster than rouge:

```
RUBY (216 kB)    CodeRay             Rouge                Pygments.rb
=> terminal           4475 kB/s            844 kB/s            433 kB/s
=> html               4285 kB/s            878 kB/s            423 kB/s
```

**Comment by vpereira** [comment]
because of the lack of support to all necessary file types and it presents a worse performance than Coderay, we are going to  close this PR. Thanks for the contribution @pk12!

**Pull request closed without merging**